react-renderer/src/engine/Optimizer.ts

```diff
@@ -531,9 +601,12 @@ const lbfgsInner = (grad_fx_k: any, ss: any[], ys: any[]): any => {
   // The length of any list should be the number of stored vectors
   const rhos = _.zip(ss, ys).map(([s, y]) => calculate_rho(s, y));
   const q_k = grad_fx_k;
-  // Note the order of alphas will be from k-1 through k-m for the push_r_forward loop  
+  // Note the order of alphas will be from k-1 through k-m for the push_r_forward loop
   // (Note that `reduce` is a left fold)
-  const [q_k_minus_m, alphas] = _.zip(rhos, ss, ys).reduce(pull_q_back, [q_k, []]);
+  const [q_k_minus_m, alphas] = (_.zip(rhos, ss, ys) as any).reduce(
```

Comment:
This is another instance of the new type errors. This is the hacky solution. @hypotext this might be a lodash thing or lack of typing in this module. I see lots of `any`s, which can be confusing to the compiler.

Reply:
I have had lodash problems with typing, but all the `any`s here are coming from `linear-algebra-js`, which doesn't have typings for its vector/matrix types. See here for when I am converting `number[]` to la.js types, which turns them into `any`. https://github.com/penrose/penrose/pull/388/files/57a0e7fe954146761270ff37f1190f34e234cea0#diff-e4cc280e366c644b840a2c549d706b1a08768a305ef0f9225323e0d622bc5c97R539

Reply:
Not sure why the diff link didn't work, but I just mean the function called `colVec` in `Optimizer`.

Reply:
`zip` will give you `undefined` sometimes (as noted in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/23305). Not clear on why exactly it's a problem here. BTW here's a site that helps you get rid of lodash https://youmightnotneed.com/lodash/, not sure if it's any good.

`any`: One way is to add types to la.js if you think it helps. If it's too much work, then some casting is perhaps inevitable.

Reply:
The reason for the new error is that your first zip statement has a type of `[number | undefined, any, any][]` perhaps due to the use the zip. I know `rhos` is definitely `number[]`, but the `any`s probably caused the extra `undefined`. I won't fix this in this PR. I think you'll need another pass on the Optimizer module to fix the types later.